src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -741,5 +771,79 @@ private void UpdateCacheExpirationTime(KeyValueWatcher changeWatcher, bool succe
 
             changeWatcher.CacheExpires = DateTimeOffset.UtcNow.Add(cacheExpirationTime);
         }
+
+        private async Task ExecuteWithFailOverPolicyAsync(Func<ConfigurationClient, Task> funcToExecute, CancellationToken cancellationToken = default)
+        {
+            IEnumerator<ConfigurationClient> clientEnumerator = _configClientProvider.GetClientEnumerator();
+
+            // Guaranteed to return at least 1 element
+            Debug.Assert(clientEnumerator.MoveNext());
+
+            while (true)
+            {
+                try
+                {
+                    cancellationToken.ThrowIfCancellationRequested();

```

Comment:
Doesn't look like `cancellationToken.ThrowIfCancellationRequested();` is needed inside the try.

Reply:
Done!